### DISCUSSION TITLE
How To Bypass Cloudflare on Second Page?

### DISCUSSION THREAD

**Discussion by khanjui** [seleniumbase/SeleniumBase#2964] [Q&A] [answered]
driver = Driver(uc=True, agent=agent)
url = "https://visa.vfsglobal.com/tur/en/pol/login"
driver.uc_open_with_reconnect(url, reconnect_time=40)

I can make the automation work on the first (login) page by using reconnect_time, but there is an additional Cloudflare check on the second page after logging in which detects the Selenium and fails to pass even if I manually click it. @mdmintz 

First page
![image](https://github.com/user-attachments/assets/39b03396-8384-40e6-9899-6750b1e3ba33)
Second page
![image](https://github.com/user-attachments/assets/17f21e77-d12f-44da-91ad-7d667d1ae167)

**Comment by mdmintz** [comment, marked as answer]
Use methods such as `sb.uc_gui_click_captcha()` to get past the CAPTCHA.
(Read the docs here: [SeleniumBase/help_docs/uc_mode.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/uc_mode.md))
Also be sure you are using `seleniumbase` `4.29.2` (or newer if available).

**Reply by khanjui** [reply]
> Use methods such as `sb.uc_gui_click_captcha()` to get past the CAPTCHA. (Read the docs here: [SeleniumBase/help_docs/uc_mode.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/uc_mode.md)) Also be sure you are using `seleniumbase` `4.29.2` (or newer if available).

I was trying everything one by one and ended up with a simple driver.disconnect() and driver.reconnect(15) being working. Thank you. Maybe a bit out of topic but is there an easy way to use/automate OTP code that comes as SMS on every login?

**Reply by mdmintz** [reply]
If you know the 16-character code that generates the 6-digit code, then there are built-in SeleniumBase methods for handling that. Here's an example test that uses MFA: [SeleniumBase/examples/test_mfa_login.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_mfa_login.py)

**Comment by MarcosGit** [comment]
good morning khanjui,

Can you share you code to bypass the first login please

**Reply by mdmintz** [reply]
```python
from seleniumbase import SB

with SB(uc=True, incognito=True) as sb:
    sb.uc_open_with_reconnect(
        "https://visa.vfsglobal.com/are/en/fra/login", 12
    )
    sb.click_if_visible("#onetrust-reject-all-handler")
    sb.uc_gui_click_captcha()
```

**Reply by MarcosGit** [reply]
i try this but not checking the claudflare checkbox

**Comment by fahadkkc** [comment]
@mdmintz Man! you're a legend.

**Reply by fahadkkc** [reply]
@mdmintz @khanjui  It successfully solves the Cloudflare CAPTCHA on the first attempt, but when I click the sign-in button, it prompts me to resolve it again. Could you please assist me with this issue?
This is my code snippet:

```
def access_website_with_solution():
    email = 'fahadali7200@gmail.com'
    password = ''
    with SB(uc=True, incognito=True) as sb:
        sb.uc_open_with_reconnect(
            "https://visa.vfsglobal.com/are/en/fra/login", 17
        )
        sb.click_if_visible("#onetrust-reject-all-handler")
        email_input = sb.find_element(By.ID, 'email')
        email_input.send_keys(email)
        password_input = sb.find_element(By.ID, 'password')
        password_input.send_keys(password)
        signin_button = sb.find_element(By.XPATH,'/html/body/app-root/div/div/app-login/section/div/div/mat-card/form/button')
        signin_button.click()
        time.sleep(5)
```
<img width="1792" alt="Screenshot 2024-10-01 at 7 59 43 PM" src="https://github.com/user-attachments/assets/b1bb2942-5624-4029-9ab9-c0a72d17863e">
<img width="1792" alt="Screenshot 2024-10-01 at 8 00 11 PM" src="https://github.com/user-attachments/assets/cfc85e0a-3e07-4e08-a2a7-c9d46dd77a02">

**Reply by mdmintz** [reply]
That situation was covered in the 3rd UC Mode video tutorial: https://www.youtube.com/watch?v=-EpZlhGWo9k

<img width="985" alt="Screenshot" src="https://github.com/user-attachments/assets/442dfb45-06b6-46fd-b1c1-4041e3b1da9d">

----

You used a regular click before encountering a CAPTCHA, which is how they detected you. Need to use `uc_click(selector)`.

If they catch you (because you didn't use stealth), then they may IP-block you, as they did:

<img width="590" alt="Screenshot" src="https://github.com/user-attachments/assets/b7c36a87-5c32-4abc-8a1b-fb87532d8f22">

**Reply by fshabbir-experlabs** [reply]
@mdmintz I repeat you're a gem. Thanks a lot for saving my life. 🫠

**Comment by jesushealth** [comment]
@khanjui How did you pass the 2nd cloudflare, my friend? I can pass the 1st one too. I get stuck on the 2nd one and I get a failed message. Can you help me?

**Reply by Jediweirdo** [reply]
If the above didn't work, try [CDP Mode](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/cdp_mode/ReadMe.md). Apparently, Cloudflare updated their stuff, so most of the time regular UC mode won't work

**Reply by jesushealth** [reply]
Yes, I was able to pass the OTP part.

**Reply by jesushealth** [reply]
Of course I sent you an e-mail

**Reply by ataberkdw** [reply]
> Yes, I was able to pass the OTP part.

Brother can you help me too I need you

**Reply by jesushealth** [reply]
@tengriG telegram hesabım yaz kardeşim

**Reply by krzysztofkub** [reply]
@jesushealth could you also help me? I cant pass cloudflare validation which opens on new page after form submition. I'm using uc mode. Works fine when manually clicking but gui captcha click fails

**Reply by jesushealth** [reply]
@krzysztofkub tg : @tengriG I can help you can send a message

**Reply by mdmintz** [reply]
People should be switching to [SeleniumBase CDP Mode](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/cdp_mode/ReadMe.md).
Regular [UC Mode](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/uc_mode.md) isn't always enough for this anymore.
